README.md

```diff
@@ -1,16 +1,171 @@
+* Define how the `map()` method works
+* Demonstrate `map()` with `Array`s
+* Demonstrate `map()` with complex data structures
+
 
 ## Introduction
 
-## SWBAT 1
+Iterating over arrays is a very common task we will find ourselves
+performing as developers. Using `for` loops, can get tedious after
+a while. Rather than using `for` loops and nesting to process data
+in lists and collections, you can take advantage a method like `map()`
+to better organize your logic into building blocks of functions,
+and then chain them to create more readable and understandable
+implementations.
+
+## Define How the `map()` Method Works
+
+`Array.prototype.map()` is an array method that iterates over all elements,
+allowing you to apply a function to each element in that array, effectively
+transforming them into something else. The result is then returned as a *new*
+array, leaving the original array intact and unmodified (but remember, **not**
+the elements we modify, necessitating the need for defensive copying). That
+last part is super important, because it either saves us from having to create
+a new array ourselves and copy stuff in there, **or** modifying the original
+elements in the array
+
+## Demonstrate `map()` with `Array`s
+
+We use `map()` when we want to perform some operation on each element in the
+collection, and "gather" the results into a new `Array`.
+
+We'll also use this as a chance to demonstrate some of the power of
+functions in JavaScript. We'll write `map()` **four times** so you can see how
+the iterator functions allow us to write more _expressive_ code.
+
+### `for` and `for...of`
+
+In this example, we are using a standard bit of iteration code. Because
+`for ` and `for...of` are _general_ functions that can be used to do lots
+of things another programmer won't be sure if the inner workings' return
+values are important or not.
+
+```js
+let students = [harryPotter, ronWeasley, hermioneGranger, ginevraWeasley];
+let patroni = []; // Patroni, sg. patronus
+
+for (const student of students) {
+  patroni.push( student.namePatronus() );
+}
+
+//=> patroni = ["Stag", "Jack Russell Terrier", "Otter", "Horse"];
+```
+
+When we write `.map()` we are saying to other programmers: expect a new array
+to come out of this thing after each element is touched! Let's use that more
+expressive function name.
+
+### `map()` and a function declaration
+
+```js
+function studentPatronusName(student) {
+  return student.namePatronus();
+}
+
+let students = [harryPotter, ronWeasley, hermioneGranger, ginevraWeasley];
+let patroni = students.map(studentPatronusName);
+//=> patroni = ["Stag", "Jack Russell Terrier", "Otter", "Horse"];
+```
+
+With `map()` we're passing a function _as an argument_. Yes, arguments can be
+things like `Number` or `String`...but, in JavaScript, **can also** be
+**work**. Very few other programming languages allow that! That's worthy of a
+Keanu "Whoa."
+
+![Whoa](https://media.giphy.com/media/ALZ1PPM20REZ2/giphy.gif)
+
```

Comment:
I think we all agreed we weren't going heavy on gifs moving forward ... I don't think this necessarily has to be taken out, but I don't think we necessarily need it either. I like Keanu! But I also wonder if the reference is a little dated.

Reply:
This probably was leftover from porting it over. I agree that gifs (basically memes), unless actually relevant to the learning process, are unnecessary.